databricks/koalas/indexes.py

```diff
@@ -360,6 +360,36 @@ def rename(self, name: Union[str, Tuple[str, ...]], inplace: bool = False):
         else:
             return Index(DataFrame(internal), scol=self._scol)
 
+    # TODO: add downcast parameter for fillna function
+    def fillna(self, value):
+        """
+        Fill NA/NaN values with the specified value.
+
+        Parameters
+        ----------
+        value : scalar
+            Scalar value to use to fill holes (e.g. 0). This value cannot be a list-likes.
+
+        Returns
+        -------
+        Index :
+            filled with value
+
+        Examples
+        --------
+        >>> ki = ks.DataFrame({'a': ['a', 'b', 'c']}, index=[1, 2, None]).index
+        >>> ki
+        Float64Index([1.0, 2.0, nan], dtype='float64')
+
+        >>> ki.fillna(0)
+        Float64Index([1.0, 2.0, 0.0], dtype='float64')
+        """
+        if not isinstance(value, (float, int, str, bool)):
+            raise TypeError("Unsupported type %s" % type(value))
+        sdf = self._internal.sdf.fillna(value)
+        result = Index(DataFrame(_InternalFrame(sdf=sdf, index_map=self._internal.index_map)))
```

Comment:
i think it will be lost name of index.

```python
>>> import databricks.koalas as ks
>>> import numpy as np

>>> kidx = ks.Index([1.0, 2.0, np.nan], name='koalas')
>>> kidx.fillna(10000)
Float64Index([1.0, 2.0, 10000.0], dtype='float64')
```

i think you can

```python
DataFrame(self._kdf._internal.copy(sdf=sdf)).index
```